tutorial/LFRic/background/LFRic_kernel.md

```diff
@@ -0,0 +1,241 @@
+# LFRic kernel
+
+[*Kernel layer in LFRic in PSyclone documentation*](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#kernel)
+
+## What kernels do
+
+LFRic kernels perform mathematical operations on [a subset of data
+points](
+https://psyclone.readthedocs.io/en/stable/kernel_layer.html#kernel-layer)
+in LFRic data objects (fields, operators and scalars) passed
+from the [algorithm](LFRic_algorithm.md) through the
+[PSy layer.](LFRic_PSy.md)
+
+Scalar values are passed to a kernel from the
+[algorithm layer](LFRic_algorithm.md) as they are. In case of fields
+and operators, however, LFRic kernels operate on a subset of degrees
+of freedom (DoFs) of these objects. Such a subset occupies a portion
+of the computational domain (e.g. an entire mesh or a partition of a
+mesh that the field's DoFs are placed on). The subsections below
+illustrate the LFRic kernel structure and how the relevant information
+is passed to a kernel.
+
+## Kernel structure
+
+In this section we use the
+[`../example1/setval_field_w0_kernel_mod.f90`](
+../example1/setval_field_w0_kernel_mod.f90)
+kernel stub to illustrate the structure of LFRic kernels (the
+[completed example](../example1/solutions/part1/setval_field_w0_kernel_mod.f90)
+is provided in the relevant [solutions](../example1/solutions/part1)).
+
+```fortran
+module setval_field_w0_kernel_mod
+
+  use argument_mod,      only: arg_type,          &
+                               GH_FIELD, GH_REAL, &
+                               GH_INC, GH_READ, CELLS
+  use fs_continuity_mod, only: W0
+  use constants_mod,     only: r_def, i_def
+  use kernel_mod,        only: kernel_type
+
+  implicit none
+
+  private
+
+  !-----------------------------------------------------------------------------
+  ! Public types
+  !-----------------------------------------------------------------------------
```

Comment:
Could we consider something else here. This comment is really unecessary and unhelpful. Clearly they are public types.

Reply:
Changed to
```
  !-----------------------------------------------------------------------------
  ! The type declaration for the kernel. Contains the metadata needed by
  ! the PSy layer.
  !-----------------------------------------------------------------------------
```
here and in all examples.